components/org.wso2.carbon.business.rules.web/src/components/TemplateGroupSelector.jsx

```diff
@@ -51,51 +51,97 @@ class TemplateGroupSelector extends React.Component {
     constructor(props) {
         super(props);
         this.state = {
-            mode: props.mode, // 'template' or 'scratch'
-            templateGroups: props.templateGroups // Available Template Groups
+            mode: this.props.match.params.mode,
+            templateGroups: []
         }
     }
 
+    componentDidMount() {
+        let that = this
+        let templateGroupsPromise = BusinessRulesUtilityFunctions.getTemplateGroups()
+        templateGroupsPromise.then(function (templateGroupsResponse) {
+            let filteredTemplateGroups = that.removeInvalidTemplateGroups(templateGroupsResponse.data[2])
+            that.setState({
+                templateGroups: filteredTemplateGroups
+            })
+        })
```

Comment:
Remove the promise variable. Instead of function use arrow function, then you can get rid of `let that = this` since the arrow functions bind parent's context to the function itself.

Reply:
The same style has been used all over the places, and I have got rid of them after enabling ESLint (which is **not** included in this PR).